code/renderers/react/src/preset.ts

```diff
@@ -0,0 +1,6 @@
+import type { StorybookConfig } from '@storybook/types';
+
+export const addons: StorybookConfig['addons'] = [
+  // Can't use path in this file due to be compiled for browser, this is a workaround
+  require.resolve('@storybook/react-dom-shim/package.json').replace('package.json', ''),
```

Comment:
I didn't test, but I don't think this works in Vite.  It does not process absolute paths.  We used to have some hackery to remove everything before the first `node_modules` in the path, but we [removed](https://github.com/storybookjs/storybook/pull/19689/files#diff-ea13e3f3305c1041c67e817923ad76d848fb562b785f94545a6455e3fac7ec03L18) that after re-working sub-addons to include both a bare and absolute key.  I wonder if we will have the same issue with this.

Reply:
Ideally, we should try to find a way to test vite pnpm projects before merging/releasing.  Now that Norbert has created a yarn pnp sandbox, perhaps we can do the same for pnpm.

Reply:
Ok, maybe @ndelangen can tell me what the right way for one addon to export another from the addons field. I was copying this pattern from some of our framework packages, but now I see they were all webpack frameworks.

Should I just be using the string '@storybook/react-dom-shim' here?

Reply:
You can look at addon-essentials to see an example of an addon including other addons.

Reply:
@IanVS, yes I notice it is using plain package names. I just want to double check with @ndelangen if that's right.

Reply:
OK, no we need to `require.resolve` everything. In this case it's fine because we aren't injecting anything from this package into Vite anyway (there's no previewAnnotations in this package).